File: array-hash/49-Group-Anagrams.go
func groupAnagrams(strs []string) [][]string {
	var results [][]string
	anagramHash := make(map[string][]string)

	for _, str := range strs {
		chars := strings.Split(str, "")
		sort.Strings(chars)
		orderedStr := strings.Join(chars, "")
		anagramHash[orderedStr] = append(anagramHash[orderedStr], str)
	}

	for _, value := range anagramHash {
		results = append(results, value)
	}

	return results
}

File: reference/array-hash/49-Group-Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagram_hash = defaultdict(list)

        for s in strs:
            ordered_str = "".join(sorted(s))  # Sort the string to get the key
            anagram_hash[ordered_str].append(s)

        return list(anagram_hash.values())